jax/interpreters/pxla.py

```diff
@@ -71,77 +72,26 @@
 Index = Union[int, slice, Tuple[Union[int, slice], ...]]
 
 
-class NoSharding:
-
-  def __eq__(self, other):
-    return isinstance(other, NoSharding)
-
-  def __repr__(self):
-    return "NoSharding()"
-
+# See the C++ code for comments.
+NoSharding = pmap_lib.NoSharding
+Chunked = pmap_lib.Chunked
+Unstacked = pmap_lib.Unstacked
 
 _UNSHARDED_INSTANCE = NoSharding()
 
-# mypy is very unhappy about taggedtuple
-if TYPE_CHECKING:
-  class Unstacked(NamedTuple):
-    size: int
-else:
-  Unstacked = taggedtuple('Unstacked', ('size',))
 
-class Chunked:
-  chunks: Tuple[int, ...]
+def make_chunked(chunks: Union[int, Sequence[int]]):
```

Comment:
Can't you subclass `pmap_lib.Chunked` and override `__init__`?

Reply:
Alternatively, I think we can just replace all uses of this with a more explicit list comprehension that constructs `Chunked` with singleton lists